test/erc20.spec.ts

```diff
@@ -1,128 +1,191 @@
-import { expect } from './setup'
-
+/* External Imports */
 import { ethers } from 'hardhat'
-import { Contract, Signer } from 'ethers'
+import { Signer, ContractFactory, BigNumber } from 'ethers'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+/* Internal Imports */
+import { expect } from './setup'
+import deploymentsInfo from '../deployments/l1/ERC20.json'
+import { ERC20 } from '../typechain/ERC20'
 
 describe('ERC20', () => {
-  let account1: Signer
-  let account2: Signer
-  let account3: Signer
-  before(async () => {
+  let ERC20: ERC20,
+    provider: JsonRpcProvider,
+    account1: Signer,
+    account2: Signer,
+    account3: Signer
+
+  const name: string = 'Some Really Cool Token Name'
+  const initialSupply: number = 10_000_000
+  const useL2: boolean = (process.env.TARGET === 'ovm')
+
+  // // For the Optimistic test file
+  // const privateKey1: string = ethers.Wallet.createRandom().privateKey
+  // const privateKey2: string = ethers.Wallet.createRandom().privateKey
+  // const privateKey3: string = ethers.Wallet.createRandom().privateKey
+
+  // Set network provider
+  if (useL2 == true) {
+    provider = new ethers.providers.JsonRpcProvider(process.env.L2_WEB3_URL)
+  } else {
+    provider = new ethers.providers.JsonRpcProvider(process.env.L1_WEB3_URL)
```

Comment:
We might want to default these values to what is in `.env.example` in case the developer forgets to copy them into `.env` locally?